### ISSUE TITLE
Add --file to image build and clean up file filtering

### ISSUE THREAD

**Issue by bufdev** [bufbuild/buf#39]
To do this consistently across sources and image, `extimage.ImageWithSpecificNames` must handle whether imports are included or not. Then, the flag can be easily added.

**Comment by willstott101** [comment]
It's very confusing that this is already in the docs ^_^

**Comment by bufdev** [comment]
Yes, our apologies, this is our fault. We will get to this soon.
